model.go

```diff
@@ -400,21 +400,24 @@ func (f *Flag) FormatPlaceHolder() string {
 	if f.Value.IsSlice() && f.Value.Tag.Sep != -1 {
 		tail += string(f.Value.Tag.Sep) + "..."
 	}
+	if f.PlaceHolder != "" {
+		return f.PlaceHolder + tail
+	}
 	if f.Default != "" {
 		if f.Value.Target.Kind() == reflect.String {
 			return strconv.Quote(f.Default) + tail
 		}
 		return f.Default + tail
 	}
-	if f.PlaceHolder != "" {
-		return f.PlaceHolder + tail
-	}
 	if f.Value.IsMap() {
 		if f.Value.Tag.MapSep != -1 {
 			tail = string(f.Value.Tag.MapSep) + "..."
 		}
 		return "KEY=VALUE" + tail
 	}
+	if f.Tag != nil && f.Tag.TypeName != "" {
```

Comment:
Can you add a test for this?

Reply:
And I must admit I am not sure when is type name passed and when not.

Reply:
Huh ok, that's a bit odd...